lib/OpenQA/Worker/Jobs.pm

```diff
@@ -694,6 +695,20 @@ sub upload_images {
     return !$tx || $tx->success;
 }
 
+sub _strip_path_prefix {
+    my ($path) = @_;
+    my ($vol, $dir, $file) = File::Spec->splitpath($path);
+    my $relpath = $file;
+    my @dirs    = File::Spec->splitdir($dir);
+    for my $d (reverse @dirs) {
```

Comment:
None taken.
As an input I have a absolute path to json file. From this path I want to remove part which is relevant only to the worker. Usually it is the same as `$prjdir=/var/lib/openqa/share`. But in case of the reported issue, this path was actually expanded symlink and completely different (`/home/user/projects/openqa-distri-opensuse`).
So in this function I split dir path and filename, then explode dir path to array of directories and then in loop try to find what is the path to the file if `$prjdir` prefix is used. Once found, this path (without $prjdir) is returned as this is the relative path which should be same on worker and webui.

Example:
input '/home/user/openqa-tests/needles/needle.json` where `/home/user/openqa-tests` is symlinked to `/var/lib/openqa/distri/tests`

in loop I try if `/var/lib/openqa/distri/needle.json`, `/var/lib/openqa/distri/needles/needle.json` exists. Which only the latter one does.
In essence I do a reverse symlink lookup where from original path I get the symlink and then strip the $prjdir.

Which is wrong, since I should strip casedir instead of $prjdir me thinks, so I need to check that again (it worked for @mnowaksuse but may be special config).